src/blob/index/bptree/core.rs

```diff
@@ -293,8 +292,11 @@ where
         self.file.read_at(&mut buf[..buf_size], leaf_offset).await?;
         let rh_size = self.header.record_header_size;
         if let Some((header, offset)) = self.read_header_buf(&buf[..buf_size], key, rh_size)? {
-            let mut headers = vec![header];
-            self.go_left(&mut headers, &buf[..buf_size], offset).await?;
+            let mut headers = vec![];
```

Comment:
Create with capacity. By default, the first `push` to the vector increases its capacity to 4 items. In our use case the most common number of items is `1`